package.json

```diff
@@ -81,14 +83,22 @@
     "react-dom": "^18.2.0",
     "rimraf": "^5.0.5",
     "semver": "^7.5.4",
+    "sherif": "^0.5.0",
     "solid-js": "^1.8.1",
     "stream-to-array": "^2.3.0",
     "ts-node": "^10.7.0",
     "tsup": "^7.2.0",
     "type-fest": "^4.5.0",
-    "typescript": "^5.0.4",
+    "typescript": "5.1.6",
```

Comment:
@lachlancollins we should probably revive the good parts of this PR:

- https://github.com/TanStack/query/pull/5640/files#diff-15c806aa509538190832852f439e9921a23bec2da81f95ed0e4bf13c14e5b160R101-R105

to test against multiple typescript versions in CI. We want to support TS 4.7, and if we code with TS 5+, we might use a feature that doesn't exist for lower versions.

Reply:
Interesting, do you know why this PR wasn't completed?

`main` was already using a combination of TS 5.1.6 (eslint) and 5.2.2 (everything else) so this was more of a consistency fix rather than a TS version bump

Reply:
> Interesting, do you know why this PR wasn't completed?

no, we did this on a live stream but then zack closed it. I don't remember if there were any issues.

> main was already using a combination of TS 5.1.6 (eslint) and 5.2.2 (everything else) so this was more of a consistency fix rather than a TS version bump

yes, I'm aware. It would still be nice I think  :)